saexec/saexec_test.go

```diff
@@ -724,3 +730,48 @@ var _ = blockstest.ModifyHeader((*blockNumSaver)(nil).store)
 func (e *blockNumSaver) store(h *types.Header) {
 	e.num = new(big.Int).Set(h.Number)
 }
+
+func TestSnapshotPersistence(t *testing.T) {
+	ctx, sut := newSUT(t, defaultHooks())
+
+	e, chain, wallet := sut.Executor, sut.chain, sut.wallet
+
+	const n = 10
+	for range n {
+		b := chain.NewBlock(t, types.Transactions{
+			wallet.SetNonceAndSign(t, 0, &types.LegacyTx{
+				To:       &common.Address{},
+				Gas:      params.TxGas,
+				GasPrice: big.NewInt(1),
+			}),
+		})
+		require.NoError(t, e.Enqueue(ctx, b), "Enqueue()")
+	}
+	last := chain.Last()
+	require.NoErrorf(t, last.WaitUntilExecuted(ctx), "%T.Last().WaitUntilExecuted()", chain)
+
+	require.NoErrorf(t, e.Close(), "%T.Close()", e)
+	// [newSUT] creates a cleanup that also calls [Executor.Close], which isn't
+	// valid usage. The simplest workaround is to just replace the quit channel
+	// so it can be closed again.
+	e.quit = make(chan struct{})
+
+	// The crux of the test is whether we can recover the EOA nonce using only a
+	// new set of snapshots, recovered from the databases.
+	conf := snapshot.Config{
+		CacheSize:  128,
+		AsyncBuild: false,
```

Comment:
Even without the `Cap` call, this test passes. Specifically, it'll start generating a snapshot in the background. I think we should be specifying `NoBuild` so that if the snap isn't available then it errors.
```suggestion
		CacheSize: 128,
		NoBuild:   true,
```